src/algorithms/tracking/TrackParamTruthInit.cc

```diff
@@ -93,10 +99,36 @@ eicrecon::TrackParamTruthInit::produce(const edm4hep::MCParticleCollection* mcpa
         // modify initial momentum to avoid bleeding truth to results when fit fails
         const auto pinit = pmag * (1.0 + m_cfg.m_momentumSmear * m_normDist(generator));
 
+        // define line surface for local position values
+        auto perigee = Acts::Surface::makeShared<Acts::PerigeeSurface>(Acts::Vector3(0,0,0));
+
+        // track particle back to transverse point-of-closest approach
+        // with respect to the defined line surface
+        auto linesurface_parameter = -(v.x*p.x + v.y*p.y)/(p.x*p.x + p.y*p.y);
+
+        auto xpca = v.x + linesurface_parameter*p.x;
+        auto ypca = v.y + linesurface_parameter*p.y;
+        auto zpca = v.z + linesurface_parameter*p.z;
+
+        Acts::Vector3 global(xpca, ypca, zpca);
+
+        // convert from global to local coordinates using the defined line surface
+        Acts::Vector2 localpos;
```

Comment:
Move to first assignment.

Reply:
I have done the suggested changes. Please check, thank you!